In [ ]:
Roll no:108806
Name: Rohith R
course name:DA
batch:geetika mam batch (9:30 to 11:30)

In [1]:
# importing the libraries
import pandas as pd
import numpy as np

### a. Build 99% Confidence Interval Using Sample Standard Deviation

In [3]:
#create a numpy array
df=np.array([[1.13],
               [1.55],
               [1.43],
               [0.92],
               [1.25],
               [1.36],
               [1.32],
               [0.85],
               [1.07],
               [1.48],
               [1.20],
               [1.33],
               [1.18],
               [1.22],
               [1.29]
              ])
df

array([[1.13],
       [1.55],
       [1.43],
       [0.92],
       [1.25],
       [1.36],
       [1.32],
       [0.85],
       [1.07],
       [1.48],
       [1.2 ],
       [1.33],
       [1.18],
       [1.22],
       [1.29]])

In [4]:
#  To find the mean in the of the data
df.mean()

1.2386666666666666

In [5]:
# To find the same standard deviation
df.std()

0.18661427836285438

In [6]:
from scipy import stats
interval=stats.norm.interval(0.99,loc=df.mean(),scale=df.std())
interval

(0.7579801399989947, 1.7193531933343384)

- import the numpy library

- Each row represents a single data point and Each column represents a variable.

- **df.mean():** function is used to compute the mean (average) of all the elements in the array. The mean is the sum of all values divided by the number of values.

- **df.std():** function computes the standard deviation of the values in the array. The standard deviation measures the amount of variation or dispersion in the dataset

- **Scipy:** uses the SciPy library to calculate a 99% confidence interval for a normal distribution

- **stats.norm.interval(0.99, loc, scale):** calculates the range of values where the true mean of the population would fall, with 99% confidence, based on the sample mean and standard deviation.

- 99 % confindance that population mean will be lies under this interval **(.7579801399989947, 1.7193531933343384)**

### b. Build 99% Confidence Interval Using Known Population Standard Deviation

In [7]:
# given population standard deviation
std=0.2 # in millions

In [8]:
from scipy import stats
interval=stats.norm.interval(0.99,loc=df.mean(),scale=std)
interval

(0.7235008059568865, 1.7538325273764466)

- This imports the norm object from the scipy.stats module

- stats.norm.interval: This is a method used to compute the confidence interval for a normal distribution.

- **Known population standard deviation (scale):** This is used to define the spread of the distribution, and it is known to be **0.2 million** characters.

- There is a very high level of confidence (99%) that the actual average number of characters a print-head can print before failure falls between **(0.7235008059568865, 1.7538325273764466)** million characters.